src/Foxy.ts

```diff
@@ -0,0 +1,62 @@
+import Handler, { Request, Info } from "./handler/Handler";
+
+
+export interface Options {
+    handlers?: Array<Handler>;
+}
+
+
+export default class Foxy {
+    handlers;
+
+    constructor(options: Options = {}) {
+        this.handlers = options.handlers || [];
+    }
+
+    addHandler(handler: Handler, index = 0): void {
+        if (this.handlers[index] == null) {
+            this.handlers.push(handler);
+        }
+        this.handlers.splice(index, 0, handler);
+    }
+
+    removeHandler(handler: Handler): void {
+        this.handlers = this.handlers.filter(h => h !== handler);
+    }
+
+    getImageURL(request: Request): Promise<string> {
```

Comment:
You might want to take a look at ES2015 proxies https://stackoverflow.com/a/42244695